server/channels/store/sqlstore/channel_store.go

```diff
@@ -2021,6 +2021,92 @@ func (s SqlChannelStore) GetChannelMembersTimezones(channelId string) ([]model.S
 	return dbMembersTimezone, nil
 }
 
+func (s SqlChannelStore) GetChannelsWithUnreadsAndWithMentions(ctx context.Context, channelIDs []string, userID string) ([]string, []string, map[string]int64, error) {
+	query := s.getQueryBuilder().Select(
+		"Channels.Id",
+		"Channels.Type",
+		"Channels.TotalMsgCount",
+		"Channels.LastPostAt",
+		"ChannelMembers.MsgCount",
+		"ChannelMembers.MentionCount",
+		"ChannelMembers.NotifyProps",
+		"ChannelMembers.LastViewedAt",
+		"Users.NotifyProps",
+	).
+		From("ChannelMembers").
+		InnerJoin("Channels ON ChannelMembers.ChannelId = Channels.Id").
+		InnerJoin("Users ON ChannelMembers.UserId = Users.Id").
```

Comment:
@agarciamontoro This is only to get the notify props of a single user so... maybe it is easier to call get user before instead of doing a join. But I am not sure of what is really better in these scenario: is it better to have on more (potentially cached) call to the database to get the user to avoid a join with another table, or is it better to join another table to avoid a database call?

Reply:
We can only answer this with confidence through testing, I'm afraid. That said, if we're interested in a single user, joining with the whole table definitely seems overkill, so I'd go with the additional call.